Common/Models/PumpManagerUI.swift

```diff
@@ -25,6 +26,12 @@ func PumpManagerHUDViewFromRawValue(_ rawValue: PumpManagerHUDViewRawValue, plug
     {
         return nil
     }
+
+    // this type casting is needed to have the mock pump HUD view display in the widget
+    if let mockPumpManager = manager as? MockPumpManager.Type {
```

Comment:
If I'm reading this right, this shouldn't be necessary. What happens if this code isn't here? What does `manager.createHUDView(rawValue: rawState)` return?

Also, not sure if this makes a difference, but the signature for this function in `PumpManagerUI` protocol and `DashPumpManager` extension is:
```
static func createHUDView(rawValue: HUDProvider.HUDViewRawState) -> LevelHUDView?
```
while the signature in `MockPumpManager` extension is:
```
static func createHUDView(rawValue: [String : Any]) -> LevelHUDView?
```
Might the subtle `rawValue` type difference be causing an issue? What happens if you change `MockPumpManager` to match?

Reply:
@darinkrauss yes, I noticed that as well and will be pushing up a LoopKit PR to make them the same. 

Part of this issue I'm experiencing is if I put in break points, the code runs without issue. When trying to print to the console from the extension it doesn't display (not sure why this is the case or how to fix it). All I really know at this point is that `PumpManagerHUDViewFromRawValue` returns `nil`.